app/src/main/java/fr/free/nrw/commons/contributions/ContributionsPresenter.java

```diff
@@ -77,10 +78,19 @@ public void saveContribution(Contribution contribution) {
             .save(contribution)
             .subscribeOn(ioThreadScheduler)
             .subscribe(() -> {
+                Constraints constraints = new Constraints.Builder()
+                    .setRequiredNetworkType(NetworkType.CONNECTED)
+                    .build();
+                OneTimeWorkRequest updatedUploadRequest = new OneTimeWorkRequest
+                    .Builder(UploadWorker.class)
+                    .setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
+                    .setConstraints(constraints)
+                    .setBackoffCriteria(BackoffPolicy.LINEAR, 10, TimeUnit.SECONDS)
```

Comment:
Is the initial delay for the first retry (10 secs) recommended? If not, have we considered the following points for choosing the initial delay?

- The initial delay should account for typical network latency and server response times or server load.
- A longer delay time would make the app look unresponsive (unless it's happening in the background).
- A shorter delay would lead to higher retry count in a short period and could put a significant load on the server, potentially leading to degraded performance.
- Frequent retries can consume device resources, including battery. If our app is constantly retrying in the background, it could impact battery life and overall device performance.


@nicolas-raoul @sivaraam Do we have a rate limit for API calls in the app to prevent Denial of Service attacks? If yes, we'll have to ensure that our retries frequency is within the limit.

Reply:
`make the app look unresponsive` -> The delay does not block the UI
`Denial of Service` -> We are still very far from that level.